rust/xaynet-server/src/state_machine/mod.rs

```diff
@@ -238,29 +246,34 @@ pub enum StateMachineInitializationError {
 }
 
 /// The state machine initializer that initializes a new state machine.
-pub struct StateMachineInitializer {
+pub struct StateMachineInitializer<C, M>
+where
+    C: CoordinatorStorage,
+    M: ModelStorage,
+{
     pet_settings: PetSettings,
     mask_settings: MaskSettings,
     model_settings: ModelSettings,
     #[cfg(feature = "model-persistence")]
     restore_settings: RestoreSettings,
 
-    redis_handle: redis::Client,
-    #[cfg(feature = "model-persistence")]
-    s3_handle: s3::Client,
+    store: Store<C, M>,
     #[cfg(feature = "metrics")]
     metrics_handle: MetricsSender,
 }
 
-impl StateMachineInitializer {
+impl<C, M> StateMachineInitializer<C, M>
```

Comment:
This module is growing. Maybe it would be worth having an `initializer.rs` module. In a follow-up PR though.

Reply:
I agree, I will add a ticket in the storage api follow up story